docs/src/tutorials/entry-tutorial.md

```diff
@@ -0,0 +1,1085 @@
+# Entry-level Tutorial on the Model Checker
+
+**Difficulty: Blue trail -- Easy**
+
+In this tutorial, we show how to turn an implementation of binary search into a
+TLA+ specification. This implementation is known to have an out-of-bounds
+error, which once existed in Java, see [Nearly All Binary Searches and
+Mergesorts are Broken][] by Joshua Bloch (2006). Our goal is not to write a
+specification of an abstract binary search algorithm. You can find such a
+specification and a proof in [Proving Safety Properties][] and [Binary search
+with a TLAPS proof][] by [Leslie Lamport][] (2019).
+
+This tutorial is written under the assumption that the reader does not have any
+knowledge of TLA+ and Apalache. Since we are not diving into protocol and
+algorithm specifications too quickly, this is a nice example to start with.
+
+Sometimes, we refer to the model checker TLC in this text. TLC is the standard
+model checker for TLC, which was introduced in the late 90es.  If you are new
+to TLA+ and want to learn more about TLC, check the [TLC][] project and the
+[TLA+ Video Course][] by Leslie Lamport.  If you are an experienced TLC user,
+you will find this tutorial helpful too, as it demonstrates the strong points
+of Apalache.
+
+We discover that the same bug not only causes the out-of-bounds error, but also
+prevents the algorithm from terminating in `log2(high - low + 1)`, as it is
+usually expected.
```

Comment:
For me, this paragraph feels out of context here. You may want to merge it with the first paragraph above, or just move it to the relevant section below.

Reply:
True. I have expanded this paragraph and merged it with the second paragraph.